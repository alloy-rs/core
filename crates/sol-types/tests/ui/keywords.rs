use alloy_sol_types::sol;

macro_rules! make {
    ($($kw:tt)*) => {$(
        sol! {
            struct $kw {
                uint $kw;
            }

            function $kw(uint $kw);
        }
    )*};
}

// Allowed
make! {
    const
    extern
    fn
    impl
    loop
    mod
    move
    mut
    pub
    ref
    trait
    unsafe
    use
    where
    async
    await
    dyn
    become
    box
    priv
    unsized
    yield
}

// Not allowed, but should be (panics on instantiation)
make! {
    crate
    self
    Self
    super
}

// Not allowed
make! {
    as
    break
    continue
    else
    enum
    false
    for
    if
    in
    let
    match
    return
    static
    struct
    true
    type
    while
    abstract
    do
    final
    macro
    override
    typeof
    virtual
    try
}

fn main() {}
