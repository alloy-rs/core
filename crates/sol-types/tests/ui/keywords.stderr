error: expected identifier, found keyword `crate`
  --> tests/ui/keywords.rs:43:5
   |
43 |     crate
   |     ^^^^^ expected identifier, found keyword

error: expected identifier, found keyword `crate`
  --> tests/ui/keywords.rs:43:5
   |
43 |     crate
   |     ^^^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct

error: expected identifier, found keyword `self`
  --> tests/ui/keywords.rs:44:5
   |
44 |     self
   |     ^^^^ expected identifier, found keyword

error: expected identifier, found keyword `self`
  --> tests/ui/keywords.rs:44:5
   |
44 |     self
   |     ^^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct

error: expected identifier, found keyword `Self`
  --> tests/ui/keywords.rs:45:5
   |
45 |     Self
   |     ^^^^ expected identifier, found keyword

error: expected identifier, found keyword `Self`
  --> tests/ui/keywords.rs:45:5
   |
45 |     Self
   |     ^^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct

error: expected identifier, found keyword `super`
  --> tests/ui/keywords.rs:46:5
   |
46 |     super
   |     ^^^^^ expected identifier, found keyword

error: expected identifier, found keyword `super`
  --> tests/ui/keywords.rs:46:5
   |
46 |     super
   |     ^^^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct

error: expected identifier, found keyword `as`
  --> tests/ui/keywords.rs:51:5
   |
51 |     as
   |     ^^ expected identifier, found keyword
   |
help: escape `as` to use it as an identifier
   |
51 |     r#as
   |     ++

error: expected identifier, found keyword `as`
  --> tests/ui/keywords.rs:51:5
   |
51 |     as
   |     ^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct
   |
help: escape `as` to use it as an identifier
   |
51 |     r#as
   |     ++

error: expected one of `>`, a const expression, lifetime, or type, found keyword `as`
  --> tests/ui/keywords.rs:51:5
   |
51 |     as
   |     ^^ expected one of `>`, a const expression, lifetime, or type

error: expected identifier, found keyword `break`
  --> tests/ui/keywords.rs:52:5
   |
52 |     break
   |     ^^^^^ expected identifier, found keyword
   |
help: escape `break` to use it as an identifier
   |
52 |     r#break
   |     ++

error: expected identifier, found keyword `break`
  --> tests/ui/keywords.rs:52:5
   |
52 |     break
   |     ^^^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct
   |
help: escape `break` to use it as an identifier
   |
52 |     r#break
   |     ++

error: invalid const generic expression
  --> tests/ui/keywords.rs:52:5
   |
52 |     break
   |     ^^^^^
   |
help: expressions must be enclosed in braces to be used as const generic arguments
   |
52 |     { break }
   |     +       +

error: expected type, found keyword `break`
  --> tests/ui/keywords.rs:52:5
   |
52 |     break
   |     ^^^^^ expected type

error: expected identifier, found keyword `continue`
  --> tests/ui/keywords.rs:53:5
   |
53 |     continue
   |     ^^^^^^^^ expected identifier, found keyword
   |
help: escape `continue` to use it as an identifier
   |
53 |     r#continue
   |     ++

error: expected identifier, found keyword `continue`
  --> tests/ui/keywords.rs:53:5
   |
53 |     continue
   |     ^^^^^^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct
   |
help: escape `continue` to use it as an identifier
   |
53 |     r#continue
   |     ++

error: invalid const generic expression
  --> tests/ui/keywords.rs:53:5
   |
53 |     continue
   |     ^^^^^^^^
   |
help: expressions must be enclosed in braces to be used as const generic arguments
   |
53 |     { continue }
   |     +          +

error: expected type, found keyword `continue`
  --> tests/ui/keywords.rs:53:5
   |
53 |     continue
   |     ^^^^^^^^ expected type

error: expected identifier, found keyword `else`
  --> tests/ui/keywords.rs:54:5
   |
54 |     else
   |     ^^^^ expected identifier, found keyword
   |
help: escape `else` to use it as an identifier
   |
54 |     r#else
   |     ++

error: expected identifier, found keyword `else`
  --> tests/ui/keywords.rs:54:5
   |
54 |     else
   |     ^^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct
   |
help: escape `else` to use it as an identifier
   |
54 |     r#else
   |     ++

error: expected one of `>`, a const expression, lifetime, or type, found keyword `else`
  --> tests/ui/keywords.rs:54:5
   |
54 |     else
   |     ^^^^ expected one of `>`, a const expression, lifetime, or type

error: expected identifier, found keyword `enum`
  --> tests/ui/keywords.rs:55:5
   |
55 |     enum
   |     ^^^^ expected identifier, found keyword
   |
help: escape `enum` to use it as an identifier
   |
55 |     r#enum
   |     ++

error: expected identifier, found keyword `enum`
  --> tests/ui/keywords.rs:55:5
   |
55 |     enum
   |     ^^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct
   |
help: escape `enum` to use it as an identifier
   |
55 |     r#enum
   |     ++

error: expected one of `>`, a const expression, lifetime, or type, found keyword `enum`
  --> tests/ui/keywords.rs:55:5
   |
55 |     enum
   |     ^^^^ expected one of `>`, a const expression, lifetime, or type

error: expected identifier, found keyword `false`
  --> tests/ui/keywords.rs:56:5
   |
56 |     false
   |     ^^^^^ expected identifier, found keyword
   |
help: escape `false` to use it as an identifier
   |
56 |     r#false
   |     ++

error: expected identifier, found keyword `false`
  --> tests/ui/keywords.rs:56:5
   |
56 |     false
   |     ^^^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct
   |
help: escape `false` to use it as an identifier
   |
56 |     r#false
   |     ++

error: expected type, found keyword `false`
  --> tests/ui/keywords.rs:56:5
   |
56 |     false
   |     ^^^^^ expected type

error: expected identifier, found keyword `for`
  --> tests/ui/keywords.rs:57:5
   |
57 |     for
   |     ^^^ expected identifier, found keyword
   |
help: escape `for` to use it as an identifier
   |
57 |     r#for
   |     ++

error: expected identifier, found keyword `for`
  --> tests/ui/keywords.rs:57:5
   |
57 |     for
   |     ^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct
   |
help: escape `for` to use it as an identifier
   |
57 |     r#for
   |     ++

error: expected `<`, found `>`
  --> tests/ui/keywords.rs:5:9
   |
5  | /         sol! {
6  | |             struct $kw {
7  | |                 uint $kw;
8  | |             }
9  | |
10 | |             function $kw(uint $kw);
11 | |         }
   | |_________^ expected `<`
...
50 | / kws! {
51 | |     as
52 | |     break
53 | |     continue
...  |
75 | |     try
76 | | }
   | |_- in this macro invocation
   |
   = note: this error originates in the macro `sol` which comes from the expansion of the macro `kws` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected identifier, found keyword `if`
  --> tests/ui/keywords.rs:58:5
   |
58 |     if
   |     ^^ expected identifier, found keyword
   |
help: escape `if` to use it as an identifier
   |
58 |     r#if
   |     ++

error: expected identifier, found keyword `if`
  --> tests/ui/keywords.rs:58:5
   |
58 |     if
   |     ^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct
   |
help: escape `if` to use it as an identifier
   |
58 |     r#if
   |     ++

error: expected one of `>`, a const expression, lifetime, or type, found keyword `if`
  --> tests/ui/keywords.rs:58:5
   |
58 |     if
   |     ^^ expected one of `>`, a const expression, lifetime, or type

error: expected identifier, found keyword `in`
  --> tests/ui/keywords.rs:59:5
   |
59 |     in
   |     ^^ expected identifier, found keyword
   |
help: escape `in` to use it as an identifier
   |
59 |     r#in
   |     ++

error: expected identifier, found keyword `in`
  --> tests/ui/keywords.rs:59:5
   |
59 |     in
   |     ^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct
   |
help: escape `in` to use it as an identifier
   |
59 |     r#in
   |     ++

error: expected one of `>`, a const expression, lifetime, or type, found keyword `in`
  --> tests/ui/keywords.rs:59:5
   |
59 |     in
   |     ^^ expected one of `>`, a const expression, lifetime, or type

error: expected identifier, found keyword `let`
  --> tests/ui/keywords.rs:60:5
   |
60 |     let
   |     ^^^ expected identifier, found keyword
   |
help: escape `let` to use it as an identifier
   |
60 |     r#let
   |     ++

error: expected identifier, found keyword `let`
  --> tests/ui/keywords.rs:60:5
   |
60 |     let
   |     ^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct
   |
help: escape `let` to use it as an identifier
   |
60 |     r#let
   |     ++

error: expected expression, found `let` statement
  --> tests/ui/keywords.rs:60:5
   |
60 |     let
   |     ^^^
   |
   = note: only supported directly in conditions of `if` and `while` expressions

error: expected one of `>`, a const expression, lifetime, or type, found keyword `let`
  --> tests/ui/keywords.rs:60:5
   |
60 |     let
   |     ^^^ expected one of `>`, a const expression, lifetime, or type

error: expected identifier, found keyword `match`
  --> tests/ui/keywords.rs:61:5
   |
61 |     match
   |     ^^^^^ expected identifier, found keyword
   |
help: escape `match` to use it as an identifier
   |
61 |     r#match
   |     ++

error: expected identifier, found keyword `match`
  --> tests/ui/keywords.rs:61:5
   |
61 |     match
   |     ^^^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct
   |
help: escape `match` to use it as an identifier
   |
61 |     r#match
   |     ++

error: expected one of `>`, a const expression, lifetime, or type, found keyword `match`
  --> tests/ui/keywords.rs:61:5
   |
61 |     match
   |     ^^^^^ expected one of `>`, a const expression, lifetime, or type

error: expected identifier, found keyword `return`
  --> tests/ui/keywords.rs:62:5
   |
62 |     return
   |     ^^^^^^ expected identifier, found keyword
   |
help: escape `return` to use it as an identifier
   |
62 |     r#return
   |     ++

error: expected identifier, found keyword `return`
  --> tests/ui/keywords.rs:62:5
   |
62 |     return
   |     ^^^^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct
   |
help: escape `return` to use it as an identifier
   |
62 |     r#return
   |     ++

error: invalid const generic expression
  --> tests/ui/keywords.rs:62:5
   |
62 |     return
   |     ^^^^^^
   |
help: expressions must be enclosed in braces to be used as const generic arguments
   |
62 |     { return }
   |     +        +

error: expected type, found keyword `return`
  --> tests/ui/keywords.rs:62:5
   |
62 |     return
   |     ^^^^^^ expected type

error: expected identifier, found keyword `static`
  --> tests/ui/keywords.rs:63:5
   |
63 |     static
   |     ^^^^^^ expected identifier, found keyword
   |
help: escape `static` to use it as an identifier
   |
63 |     r#static
   |     ++

error: expected identifier, found keyword `static`
  --> tests/ui/keywords.rs:63:5
   |
63 |     static
   |     ^^^^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct
   |
help: escape `static` to use it as an identifier
   |
63 |     r#static
   |     ++

error: expected one of `>`, a const expression, lifetime, or type, found keyword `static`
  --> tests/ui/keywords.rs:63:5
   |
63 |     static
   |     ^^^^^^ expected one of `>`, a const expression, lifetime, or type

error: expected identifier, found keyword `struct`
  --> tests/ui/keywords.rs:64:5
   |
64 |     struct
   |     ^^^^^^ expected identifier, found keyword
   |
help: escape `struct` to use it as an identifier
   |
64 |     r#struct
   |     ++

error: expected identifier, found keyword `struct`
  --> tests/ui/keywords.rs:64:5
   |
64 |     struct
   |     ^^^^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct
   |
help: escape `struct` to use it as an identifier
   |
64 |     r#struct
   |     ++

error: expected one of `>`, a const expression, lifetime, or type, found keyword `struct`
  --> tests/ui/keywords.rs:64:5
   |
64 |     struct
   |     ^^^^^^ expected one of `>`, a const expression, lifetime, or type

error: expected identifier, found keyword `true`
  --> tests/ui/keywords.rs:65:5
   |
65 |     true
   |     ^^^^ expected identifier, found keyword
   |
help: escape `true` to use it as an identifier
   |
65 |     r#true
   |     ++

error: expected identifier, found keyword `true`
  --> tests/ui/keywords.rs:65:5
   |
65 |     true
   |     ^^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct
   |
help: escape `true` to use it as an identifier
   |
65 |     r#true
   |     ++

error: expected type, found keyword `true`
  --> tests/ui/keywords.rs:65:5
   |
65 |     true
   |     ^^^^ expected type

error: expected identifier, found keyword `type`
  --> tests/ui/keywords.rs:66:5
   |
66 |     type
   |     ^^^^ expected identifier, found keyword
   |
help: escape `type` to use it as an identifier
   |
66 |     r#type
   |     ++

error: expected identifier, found keyword `type`
  --> tests/ui/keywords.rs:66:5
   |
66 |     type
   |     ^^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct
   |
help: escape `type` to use it as an identifier
   |
66 |     r#type
   |     ++

error: expected one of `>`, a const expression, lifetime, or type, found keyword `type`
  --> tests/ui/keywords.rs:66:5
   |
66 |     type
   |     ^^^^ expected one of `>`, a const expression, lifetime, or type

error: expected identifier, found keyword `while`
  --> tests/ui/keywords.rs:67:5
   |
67 |     while
   |     ^^^^^ expected identifier, found keyword
   |
help: escape `while` to use it as an identifier
   |
67 |     r#while
   |     ++

error: expected identifier, found keyword `while`
  --> tests/ui/keywords.rs:67:5
   |
67 |     while
   |     ^^^^^
   |     |
   |     expected identifier, found keyword
   |     while parsing this struct
   |
help: escape `while` to use it as an identifier
   |
67 |     r#while
   |     ++

error: expected one of `>`, a const expression, lifetime, or type, found keyword `while`
  --> tests/ui/keywords.rs:67:5
   |
67 |     while
   |     ^^^^^ expected one of `>`, a const expression, lifetime, or type

error: expected identifier, found reserved keyword `abstract`
  --> tests/ui/keywords.rs:68:5
   |
68 |     abstract
   |     ^^^^^^^^ expected identifier, found reserved keyword
   |
help: escape `abstract` to use it as an identifier
   |
68 |     r#abstract
   |     ++

error: expected identifier, found reserved keyword `abstract`
  --> tests/ui/keywords.rs:68:5
   |
68 |     abstract
   |     ^^^^^^^^
   |     |
   |     expected identifier, found reserved keyword
   |     while parsing this struct
   |
help: escape `abstract` to use it as an identifier
   |
68 |     r#abstract
   |     ++

error: expected one of `>`, a const expression, lifetime, or type, found reserved keyword `abstract`
  --> tests/ui/keywords.rs:68:5
   |
68 |     abstract
   |     ^^^^^^^^ expected one of `>`, a const expression, lifetime, or type

error: expected identifier, found reserved keyword `do`
  --> tests/ui/keywords.rs:69:5
   |
69 |     do
   |     ^^ expected identifier, found reserved keyword
   |
help: escape `do` to use it as an identifier
   |
69 |     r#do
   |     ++

error: expected identifier, found reserved keyword `do`
  --> tests/ui/keywords.rs:69:5
   |
69 |     do
   |     ^^
   |     |
   |     expected identifier, found reserved keyword
   |     while parsing this struct
   |
help: escape `do` to use it as an identifier
   |
69 |     r#do
   |     ++

error: expected one of `>`, a const expression, lifetime, or type, found reserved keyword `do`
  --> tests/ui/keywords.rs:69:5
   |
69 |     do
   |     ^^ expected one of `>`, a const expression, lifetime, or type

error: expected identifier, found reserved keyword `final`
  --> tests/ui/keywords.rs:70:5
   |
70 |     final
   |     ^^^^^ expected identifier, found reserved keyword
   |
help: escape `final` to use it as an identifier
   |
70 |     r#final
   |     ++

error: expected identifier, found reserved keyword `final`
  --> tests/ui/keywords.rs:70:5
   |
70 |     final
   |     ^^^^^
   |     |
   |     expected identifier, found reserved keyword
   |     while parsing this struct
   |
help: escape `final` to use it as an identifier
   |
70 |     r#final
   |     ++

error: expected one of `>`, a const expression, lifetime, or type, found reserved keyword `final`
  --> tests/ui/keywords.rs:70:5
   |
70 |     final
   |     ^^^^^ expected one of `>`, a const expression, lifetime, or type

error: expected identifier, found reserved keyword `macro`
  --> tests/ui/keywords.rs:71:5
   |
71 |     macro
   |     ^^^^^ expected identifier, found reserved keyword
   |
help: escape `macro` to use it as an identifier
   |
71 |     r#macro
   |     ++

error: expected identifier, found reserved keyword `macro`
  --> tests/ui/keywords.rs:71:5
   |
71 |     macro
   |     ^^^^^
   |     |
   |     expected identifier, found reserved keyword
   |     while parsing this struct
   |
help: escape `macro` to use it as an identifier
   |
71 |     r#macro
   |     ++

error: expected one of `>`, a const expression, lifetime, or type, found reserved keyword `macro`
  --> tests/ui/keywords.rs:71:5
   |
71 |     macro
   |     ^^^^^ expected one of `>`, a const expression, lifetime, or type

error: expected identifier, found reserved keyword `override`
  --> tests/ui/keywords.rs:72:5
   |
72 |     override
   |     ^^^^^^^^ expected identifier, found reserved keyword
   |
help: escape `override` to use it as an identifier
   |
72 |     r#override
   |     ++

error: expected identifier, found reserved keyword `override`
  --> tests/ui/keywords.rs:72:5
   |
72 |     override
   |     ^^^^^^^^
   |     |
   |     expected identifier, found reserved keyword
   |     while parsing this struct
   |
help: escape `override` to use it as an identifier
   |
72 |     r#override
   |     ++

error: expected one of `>`, a const expression, lifetime, or type, found reserved keyword `override`
  --> tests/ui/keywords.rs:72:5
   |
72 |     override
   |     ^^^^^^^^ expected one of `>`, a const expression, lifetime, or type

error: expected identifier, found reserved keyword `typeof`
  --> tests/ui/keywords.rs:73:5
   |
73 |     typeof
   |     ^^^^^^ expected identifier, found reserved keyword
   |
help: escape `typeof` to use it as an identifier
   |
73 |     r#typeof
   |     ++

error: expected identifier, found reserved keyword `typeof`
  --> tests/ui/keywords.rs:73:5
   |
73 |     typeof
   |     ^^^^^^
   |     |
   |     expected identifier, found reserved keyword
   |     while parsing this struct
   |
help: escape `typeof` to use it as an identifier
   |
73 |     r#typeof
   |     ++

error: expected `(`, found `>`
  --> tests/ui/keywords.rs:5:9
   |
5  | /         sol! {
6  | |             struct $kw {
7  | |                 uint $kw;
8  | |             }
9  | |
10 | |             function $kw(uint $kw);
11 | |         }
   | |_________^ unexpected token
...
50 | / kws! {
51 | |     as
52 | |     break
53 | |     continue
...  |
73 | |     typeof
   | |           - expected `(`
74 | |     virtual
75 | |     try
76 | | }
   | |_- in this macro invocation
   |
   = note: this error originates in the macro `sol` which comes from the expansion of the macro `kws` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected identifier, found reserved keyword `virtual`
  --> tests/ui/keywords.rs:74:5
   |
74 |     virtual
   |     ^^^^^^^ expected identifier, found reserved keyword
   |
help: escape `virtual` to use it as an identifier
   |
74 |     r#virtual
   |     ++

error: expected identifier, found reserved keyword `virtual`
  --> tests/ui/keywords.rs:74:5
   |
74 |     virtual
   |     ^^^^^^^
   |     |
   |     expected identifier, found reserved keyword
   |     while parsing this struct
   |
help: escape `virtual` to use it as an identifier
   |
74 |     r#virtual
   |     ++

error: expected one of `>`, a const expression, lifetime, or type, found reserved keyword `virtual`
  --> tests/ui/keywords.rs:74:5
   |
74 |     virtual
   |     ^^^^^^^ expected one of `>`, a const expression, lifetime, or type

error: expected identifier, found reserved keyword `try`
  --> tests/ui/keywords.rs:75:5
   |
75 |     try
   |     ^^^ expected identifier, found reserved keyword
   |
help: escape `try` to use it as an identifier
   |
75 |     r#try
   |     ++

error: expected identifier, found reserved keyword `try`
  --> tests/ui/keywords.rs:75:5
   |
75 |     try
   |     ^^^
   |     |
   |     expected identifier, found reserved keyword
   |     while parsing this struct
   |
help: escape `try` to use it as an identifier
   |
75 |     r#try
   |     ++

error: expected one of `>`, a const expression, lifetime, or type, found reserved keyword `try`
  --> tests/ui/keywords.rs:75:5
   |
75 |     try
   |     ^^^ expected one of `>`, a const expression, lifetime, or type

error[E0433]: failed to resolve: there are too many leading `super` keywords
  --> tests/ui/keywords.rs:5:9
   |
5  | /         sol! {
6  | |             struct $kw {
7  | |                 uint $kw;
8  | |             }
9  | |
10 | |             function $kw(uint $kw);
11 | |         }
   | |_________^ there are too many leading `super` keywords

error[E0433]: failed to resolve: there are too many leading `super` keywords
  --> tests/ui/keywords.rs:46:5
   |
46 |     super
   |     ^^^^^ there are too many leading `super` keywords

error[E0573]: expected type, found module `crate`
  --> tests/ui/keywords.rs:5:9
   |
5  | /         sol! {
6  | |             struct $kw {
7  | |                 uint $kw;
8  | |             }
9  | |
10 | |             function $kw(uint $kw);
11 | |         }
   | |_________^ not a type
...
42 | / kws! {
43 | |     crate
   | |     ----- you might have meant to refer to this struct
44 | |     self
45 | |     Self
46 | |     super
47 | | }
   | |_- in this macro invocation
   |
   = note: this error originates in the derive macro `Clone` which comes from the expansion of the macro `kws` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0574]: expected struct, variant or union type, found module `crate`
  --> tests/ui/keywords.rs:5:9
   |
5  | /         sol! {
6  | |             struct $kw {
7  | |                 uint $kw;
8  | |             }
9  | |
10 | |             function $kw(uint $kw);
11 | |         }
   | |_________^ not a struct, variant or union type
...
42 | / kws! {
43 | |     crate
   | |     ----- you might have meant to refer to this struct
44 | |     self
45 | |     Self
46 | |     super
47 | | }
   | |_- in this macro invocation
   |
   = note: this error originates in the derive macro `Clone` which comes from the expansion of the macro `kws` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0573]: expected type, found module `crate`
  --> tests/ui/keywords.rs:43:5
   |
43 |     crate
   |     ^^^^^
   |     |
   |     you might have meant to refer to this struct
   |     not a type
   |
help: consider importing this struct instead
   |
1  + use crate::crate;
   |

error[E0573]: expected type, found module `self`
  --> tests/ui/keywords.rs:5:9
   |
5  | /         sol! {
6  | |             struct $kw {
7  | |                 uint $kw;
8  | |             }
9  | |
10 | |             function $kw(uint $kw);
11 | |         }
   | |         ^ help: a self type with a similar name exists: `Self`
   | |_________|
   |
...
42 | / kws! {
43 | |     crate
44 | |     self
45 | |     Self
46 | |     super
47 | | }
   | |_- in this macro invocation
   |
   = note: this error originates in the derive macro `Clone` which comes from the expansion of the macro `kws` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0574]: expected struct, variant or union type, found module `self`
  --> tests/ui/keywords.rs:5:9
   |
5  | /         sol! {
6  | |             struct $kw {
7  | |                 uint $kw;
8  | |             }
9  | |
10 | |             function $kw(uint $kw);
11 | |         }
   | |         ^ help: a self type with a similar name exists: `Self`
   | |_________|
   |
...
42 | / kws! {
43 | |     crate
44 | |     self
45 | |     Self
46 | |     super
47 | | }
   | |_- in this macro invocation
   |
   = note: this error originates in the derive macro `Clone` which comes from the expansion of the macro `kws` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0573]: expected type, found module `self`
  --> tests/ui/keywords.rs:44:5
   |
44 |     self
   |     ^^^^
   |
help: a self type with a similar name exists
   |
44 |     Self
   |
help: consider importing this struct instead
   |
1  + use crate::self;
   |

error[E0425]: cannot find value `value` in this scope
  --> tests/ui/keywords.rs:5:9
   |
5  | /         sol! {
6  | |             struct $kw {
7  | |                 uint $kw;
8  | |             }
9  | |
10 | |             function $kw(uint $kw);
11 | |         }
   | |_________^ not found in this scope
...
50 | / kws! {
51 | |     as
52 | |     break
53 | |     continue
...  |
75 | |     try
76 | | }
   | |_- in this macro invocation
   |
   = note: this error originates in the macro `sol` which comes from the expansion of the macro `kws` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `Self` is not valid in the self type of an impl block
  --> tests/ui/keywords.rs:5:9
   |
5  | /         sol! {
6  | |             struct $kw {
7  | |                 uint $kw;
8  | |             }
9  | |
10 | |             function $kw(uint $kw);
11 | |         }
   | |_________^
...
42 | / kws! {
43 | |     crate
44 | |     self
45 | |     Self
46 | |     super
47 | | }
   | |_- in this macro invocation
   |
   = note: replace `Self` with a different type
   = note: this error originates in the derive macro `Clone` which comes from the expansion of the macro `kws` (in Nightly builds, run with -Z macro-backtrace for more info)

error: `Self` is not valid in the self type of an impl block
  --> tests/ui/keywords.rs:45:5
   |
45 |     Self
   |     ^^^^
   |
   = note: replace `Self` with a different type

error[E0747]: constant provided when a type was expected
  --> tests/ui/keywords.rs:52:5
   |
52 |     break
   |     ^^^^^

error[E0747]: constant provided when a type was expected
  --> tests/ui/keywords.rs:53:5
   |
53 |     continue
   |     ^^^^^^^^

error[E0747]: constant provided when a type was expected
  --> tests/ui/keywords.rs:56:5
   |
56 |     false
   |     ^^^^^

error[E0747]: constant provided when a type was expected
  --> tests/ui/keywords.rs:62:5
   |
62 |     return
   |     ^^^^^^

error[E0747]: constant provided when a type was expected
  --> tests/ui/keywords.rs:65:5
   |
65 |     true
   |     ^^^^
